enginetest/queries/join_queries.go

```diff
@@ -520,6 +520,54 @@ inner join pq on true order by 1,2,3,4,5,6,7,8 limit 5;`,
 	},
 }
 
+var JoinScripts = []ScriptTest{
```

Comment:
not sure we need this, simple bug deserves a simple repro? the OR forces an inner join
```
SELECT (
  SELECT 1 FROM (
    SELECT x FROM xy
    INNER JOIN uv ON (x = u OR y = v)
   ) r
) AS s FROM xy
```

Reply:
oooh!! this is great. I just confirmed this repros the issue. Thanks for pointing this out; much, much simpler repro query than what I had in there before.